onnxruntime/core/providers/nnapi/nnapi_builtin/builders/shaper.cc

```diff
@@ -0,0 +1,375 @@
+#include "core/providers/nnapi/nnapi_builtin/nnapi_lib/NeuralNetworksWrapper.h"
+#include "helper.h"
+#include "shaper.h"
+
+using std::string;
+using std::vector;
+
+void Shaper::Conv(const std::string& input_name,
+                  const std::string& weight_name,
```

Comment:
Does this do the same calculations as onnxruntime::ConvAttributes::InferOutputShape? Or is it a simplified version?

Reply:
I have a TODO to switch all the shape inferencing to existing API, but there will be some changes need since the current code supports both nchw and nhwc, will need some code to re-arrange the input and output shape